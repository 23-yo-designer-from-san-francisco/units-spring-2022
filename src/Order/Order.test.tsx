import React from 'react';
import {OrderComponent} from './Order';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Order.tsx', () => {
	it('Handles null order', () => {
		const order = null;
		const wrapper = shallow(<OrderComponent order={order}/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('Renders order with items: null', () => {
		const order = {
			id: 100,
			date: 1647541075867,
			shop: 'МАГАЗИН НА ДИВАНЕ',
			items: null,
		};
		const wrapper = shallow(<OrderComponent order={order}/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('Handles incomplete order', () => {
		const order = {
			id: 100,
			items: [],
		};
		const wrapper = shallow(<OrderComponent order={order}/>);
		expect(wrapper.getElement()).toBeNull();
	});

	it('Renders empty order', () => {
		const order = {
			id: 1,
			date: 1647541075867,
			shop: 'AZOT',
			items: [],
		};
		const wrapper = shallow(<OrderComponent order={order}/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('Renders correct order', () => {
		const order = {
			id: 1,
			date: 1647541075867,
			shop: 'SHOP',
			items: [
				'айфончик',
				'гэлакси телефон'
			],
		};
		const wrapper = shallow(<OrderComponent order={order}/>);
		expect(wrapper).toMatchSnapshot();
	});
});