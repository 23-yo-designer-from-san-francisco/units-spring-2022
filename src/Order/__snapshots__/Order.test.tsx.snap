// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.tsx Handles incomplete order 1`] = `""`;

exports[`Order.tsx Handles null order 1`] = `""`;

exports[`Order.tsx Renders correct order 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      SHOP
    </span>
    <span
      id="date"
    >
      17 марта, чт, 2022 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      айфончик
    </div>
    <div
      className="Order-item"
      key="1"
    >
      гэлакси телефон
    </div>
  </div>
</div>
`;

exports[`Order.tsx Renders empty order 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      AZOT
    </span>
    <span
      id="date"
    >
      17 марта, чт, 2022 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div>
      Заказ пуст...
    </div>
  </div>
</div>
`;

exports[`Order.tsx Renders order with items: null 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      МАГАЗИН НА ДИВАНЕ
    </span>
    <span
      id="date"
    >
      17 марта, чт, 2022 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div>
      Заказ пуст...
    </div>
  </div>
</div>
`;

exports[`Order.tsx order with dict items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      VK )
    </span>
    <span
      id="date"
    >
      17 марта, чт, 2022 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div>
      Заказ пуст...
    </div>
  </div>
</div>
`;
